go.mod: update golang.org/x deps

RaduBerinde opened the pull request:
It's a good idea to keep these dependencies up to date, especially
after we upgrade the go version.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143678)
<!-- Reviewable:end -->

RaduBerinde commented:
TFTR!

bors r+

RaduBerinde commented:
Actually, there is a Pebble bump in bors, I'll rebase after that is in.

bors r-

RaduBerinde committed:
go.mod: update golang.org/x deps

It's a good idea to keep these dependencies up to date, especially
after we upgrade the go version.

Epic: none
Release note: None

RaduBerinde commented:
bors r+
